x-pack/plugins/endpoint/public/applications/endpoint/view/alerts/resolver.tsx

```diff
@@ -12,21 +12,26 @@ import { Resolver } from '../../../../embeddables/resolver/view';
 import { EndpointPluginServices } from '../../../../plugin';
 import { LegacyEndpointEvent } from '../../../../../common/types';
 import { storeFactory } from '../../../../embeddables/resolver/store';
+import { useAlertListSelector } from './hooks/use_alerts_selector';
+import * as selectors from '../../store/alerts/selectors';
 
 export const AlertDetailResolver = styled(
-  React.memo(
-    ({ className, selectedEvent }: { className?: string; selectedEvent?: LegacyEndpointEvent }) => {
-      const context = useKibana<EndpointPluginServices>();
-      const { store } = storeFactory(context);
-      return (
-        <div className={className} data-test-subj="alertResolver" data-testid="alertResolver">
-          <Provider store={store}>
-            <Resolver selectedEvent={selectedEvent} />
-          </Provider>
-        </div>
-      );
+  React.memo(({ className }: { className?: string }) => {
+    const alertDetailsData = useAlertListSelector(selectors.selectedAlertDetailsData);
+    if (alertDetailsData === undefined) {
+      return null;
     }
-  )
+    const context = useKibana<EndpointPluginServices>();
+    const { store } = storeFactory(context);
+
+    return (
+      <div className={className} data-test-subj="alertResolver" data-testid="alertResolver">
+        <Provider store={store}>
+          <Resolver selectedEvent={(alertDetailsData as unknown) as LegacyEndpointEvent} />
```

Comment:
why not use the predicate that you removed https://github.com/elastic/kibana/pull/58412/files#diff-e42c64f81c4fa55ac260536744187224L99